beets/ui/__init__.py

```diff
@@ -1057,43 +1048,149 @@ def _open_library(config):
+        }
+
+    def list_commands(self, ctx):
+        return self._commands(ctx).keys()
+
+    def get_command(self, ctx, name):
+        # Special case for the `config --edit` command: bypass loading config
+        # so that an invalid configuration does not prevent the editor from
+        # starting.
+        if ctx.args and ctx.args[0] == 'config' and \
+           ('-e' in ctx.args or '--edit' in ctx.args):
+            from beets.ui.commands import config_cmd
+            return config_cmd
+
+        return self._commands(ctx).get(name)
+
+
+def album_option(f):
+    """Add a -a/--album option to match albums instead of tracks.
+
+    If used then the format option can auto-detect whether we're setting
+    the format for items or albums.
+    Sets the album property on the options extracted from the CLI.
+    """
+    return click.option('-a', '--album', is_eager=True, is_flag=True,
+                        help='match albums instead of tracks')(f)
+
+
+def _set_format(target, fmt):
+    config[target._format_config_key].set(fmt)
+
+
+def path_option(f):
+    """Add a -p/--path option to display the path instead of the default
+    format.
+
+    By default this affects both items and albums. If album_option()
+    is used then the target will be autodetected.
+
+    Sets the format property to u'$path' on the options extracted from the
+    CLI.
+    """
+    def callback(ctx, param, value):
+        if not value:
+            return
+
+        if 'album' in ctx.params:
+            if ctx.params['album']:
+                _set_format(library.Album, '$path')
+            else:
+                _set_format(library.Item, '$path')
+        else:
+            _set_format(library.Item, '$path')
+            _set_format(library.Album, '$path')
+
+    return click.option('-p', '--path', callback=callback, is_flag=True,
+                        help='print paths for matched items or albums')(f)
+
+
+def format_option(flags=('-f', '--format'), target=None):
+    def callback(ctx, param, value):
+        if not value:
+            return
+
+        value = unicode(value)
+
+        _target = target
+        if _target:
+            if _target in ('item', 'album'):
+                _target = {'item': library.Item,
+                           'album': library.Album}[_target]
+            _set_format(_target, value)
+        else:
+            if 'album' in ctx.params:
+                if ctx.params['album']:
+                    _set_format(library.Album, value)
+                else:
+                    _set_format(library.Item, value)
+            else:
+                _set_format(library.Item, value)
+                _set_format(library.Album, value)
+
+    def decorator(f):
+        return click.option(*flags, callback=callback, expose_value=False,
+                            help='print with custom format')(f)
+
+    return decorator
+
+
+def all_common_options(f):
+    return album_option(path_option(format_option()(f)))
```

Comment:
Looks like `list` is the only consumer of this shorthand—maybe it's not critical.